lib/sgtn-client/api/base.rb

```diff
@@ -0,0 +1,99 @@
+# 
+#  Copyright 2019-2022 VMware, Inc.
+#  SPDX-License-Identifier: EPL-2.0
+#
+
+require 'multi_json'
+
+module SgtnClient
+
+  class Base
+    class << self
+
+      def getTranslation(component, key, locale)
+        locale = fallback_locale(locale)
```

Comment:
it's conflict with compareSource which has the same call.

Reply:
Just update fallback_locale to add another argument `remoteSource=false:`
See https://github.com/vmware/singleton/blob/55ec22c26b183ea31b2176df4318dabde7c4a7ac/lib/sgtn-client/util/locale-util.rb#L15
When you call this in compareSource, use get_cs(component, default_locale, true).

Reply:
this(get_cs(component, default_locale, true)) will cause get_cs unclear, it will return uncertain value depending on the locale is the source locale or not, that's also why we should keep fallback_locale in the base.rb file

Reply:
current get_cs is for getting data from cache by the locale directly, so locale_fallback should be put out of it.